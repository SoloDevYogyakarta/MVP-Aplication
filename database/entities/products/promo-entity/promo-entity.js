"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.productPromoEntity = exports.ProductPromoEntity = void 0;
const nanoid_1 = require("nanoid");
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class ProductPromoEntity extends base_entity_1.BaseEntity {
}
exports.ProductPromoEntity = ProductPromoEntity;
exports.productPromoEntity = entity_1.sequelize.define('PROMO', {
    public_id: sequelize_1.DataTypes.STRING,
    main_stock: sequelize_1.DataTypes.INTEGER,
    reverse_stock: sequelize_1.DataTypes.INTEGER,
    value: sequelize_1.DataTypes.INTEGER,
    start_time: sequelize_1.DataTypes.DATE,
    end_time: sequelize_1.DataTypes.DATE,
    product_id: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'PROMO',
    schema: 'PRODUCTS',
    hooks: {
        beforeCreate(attributes, options) {
            const instance = attributes;
            instance.public_id = (0, nanoid_1.nanoid)();
        },
    },
});
//# sourceMappingURL=promo-entity.js.map