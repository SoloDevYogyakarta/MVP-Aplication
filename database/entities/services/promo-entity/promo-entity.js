"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.promoEntity = exports.PromoEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class PromoEntity extends base_entity_1.BaseEntity {
}
exports.PromoEntity = PromoEntity;
exports.promoEntity = entity_1.sequelize.define('PROMO', {
    name: sequelize_1.DataTypes.STRING,
    desc: sequelize_1.DataTypes.TEXT,
    image: sequelize_1.DataTypes.STRING,
    price: sequelize_1.DataTypes.INTEGER,
    discount: sequelize_1.DataTypes.INTEGER,
    start_time: sequelize_1.DataTypes.DATE,
    end_time: sequelize_1.DataTypes.DATE,
    user_id: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'PROMO',
    schema: 'SERVICES',
});
//# sourceMappingURL=promo-entity.js.map