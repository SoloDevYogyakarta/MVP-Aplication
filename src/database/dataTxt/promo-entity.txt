{"id":52,"value":38,"main_stock":1,"reverse_stock":5,"start_time":"2023-10-26T07:17:34.513Z","end_time":"2023-10-26T07:17:34.513Z","product_id":"_yyHA8PJaUF7t1wRhhLbB","updatedAt":"2023-10-25T07:17:34.516Z","createdAt":"2023-10-25T07:17:34.516Z","public_id":"jzJJgELrng1BP2ku93_VB"}