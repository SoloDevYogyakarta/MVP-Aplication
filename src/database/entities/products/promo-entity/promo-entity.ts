import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class ProductPromoEntity extends BaseEntity {
  value: number;
  main_stock: number;
  reverse_stock: number;
  start_time: Date;
  end_time: Date;
  product_id: string;
}

export type ProductPromoCreationAttribute = Optional<ProductPromoEntity, 'id'>;
export interface ProductPromoInstance
  extends Model<ProductPromoCreationAttribute, ProductPromoEntity>,
    ProductPromoEntity {}
export const productPromoEntity = sequelize.define<ProductPromoInstance>(
  'PROMO',
  {
    public_id: DataTypes.STRING,
    main_stock: DataTypes.INTEGER,
    reverse_stock: DataTypes.INTEGER,
    value: DataTypes.INTEGER,
    start_time: DataTypes.DATE,
    end_time: DataTypes.DATE,
    product_id: DataTypes.STRING,
  } as ModelAttributes<ProductPromoInstance>,
  {
    tableName: 'PROMO',
    schema: 'PRODUCTS',
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
