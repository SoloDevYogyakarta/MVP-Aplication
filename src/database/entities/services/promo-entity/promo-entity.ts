import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class PromoEntity extends BaseEntity {
  name: string;
  desc: string;
  image: string;
  price: number;
  discount: number;
  start_time: Date;
  end_time: Date;
  user_id: number;
}

export type PromoCreationAttribute = Optional<PromoEntity, 'id'>;
export interface PromoInstance
  extends Model<PromoCreationAttribute, PromoEntity>,
    PromoEntity {}
export const promoEntity = sequelize.define<PromoInstance>(
  'PROMO',
  {
    name: DataTypes.STRING,
    desc: DataTypes.TEXT,
    image: DataTypes.STRING,
    price: DataTypes.INTEGER,
    discount: DataTypes.INTEGER,
    start_time: DataTypes.DATE,
    end_time: DataTypes.DATE,
    user_id: DataTypes.STRING,
  } as ModelAttributes<PromoInstance>,
  {
    tableName: 'PROMO',
    schema: 'SERVICES',
  },
);
