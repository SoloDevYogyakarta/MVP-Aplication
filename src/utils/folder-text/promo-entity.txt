{"id":82,"name":"Luxurious Wooden Keyboard","desc":"Andy shoes are designed to keeping in mind durability as well as trends, the most stylish range of shoes & sandals","image":"https://loremflickr.com/640/480?lock=8599795125977088","price":27,"discount":34,"user_id":873,"updatedAt":"2023-11-06T14:36:11.435Z","createdAt":"2023-11-06T14:36:11.435Z","start_time":null,"end_time":null}