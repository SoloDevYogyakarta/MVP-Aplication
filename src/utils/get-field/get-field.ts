import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
