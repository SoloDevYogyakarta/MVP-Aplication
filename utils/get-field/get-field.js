"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map