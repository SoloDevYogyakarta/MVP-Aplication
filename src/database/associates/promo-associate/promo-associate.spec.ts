import { ModelCtor } from 'sequelize';
import { ProductPromoInstance } from '../../../database/entities/products/promo-entity/promo-entity';
import { getField } from '../../../utils/get-field/get-field';
import {
  productPromoAssciate,
  productPromoAttribute,
  productPromoInclude,
} from './promo-associate';

describe('ProductPromoAssociate', () => {
  let product_id!: string;
  let entity: ModelCtor<ProductPromoInstance>;

  beforeEach(() => {
    entity = productPromoAssciate;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    product_id = getField('basic-http-entity').product_id;
  } catch (err) {
    // empty
  }

  if (product_id) {
    it('findOne with relationship', async () => {
      const findOne = await entity.findOne({
        where: { product_id },
        attributes: productPromoAttribute,
        include: productPromoInclude,
      });
      expect(findOne.product_id).toEqual(product_id);
    });

    it('findAll with relationship', async () => {
      const findAll = await entity.findAll({
        attributes: productPromoAttribute,
        include: productPromoInclude,
      });
      expect(findAll.length).not.toEqual(0);
    });
  }
});
