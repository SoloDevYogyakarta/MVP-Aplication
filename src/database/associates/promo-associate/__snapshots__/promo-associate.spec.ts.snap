// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPromoAssociate render correctly 1`] = `[Function]`;
